pkg/operator/minio/controller.go

```diff
@@ -103,13 +103,22 @@ func (c *MinioController) makeMinioHeadlessService(name, namespace string, spec
 			Ports:     []v1.ServicePort{{Port: minioPort}},
 			ClusterIP: v1.ClusterIPNone,
 		},
-	})
+	}
 	k8sutil.SetOwnerRef(c.context.Clientset, namespace, &svc.ObjectMeta, &ownerRef)
 
-	return svc, err
+	svc, err := c.context.Clientset.CoreV1().Services(namespace).Create(svc)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return nil, err
```

Comment:
Do we need to log, or return a composite error such as this? It's been helpful to debug the source of errors with that pattern.
`return nil, fmt.Errorf("failed to create service. %+v", err)`

Reply:
I have gone ahead and made the `make...()` funcs return full errors and the using part to then log them out.